### ISSUE TITLE
Improved library game display

### ISSUE THREAD

**Issue by Democide** [itchio/itch#2261]
Currently the main library view shows to horizontal lines, one with OWNED and one with INSTALLED games:

![image](https://user-images.githubusercontent.com/7945562/53158510-109b8300-35c4-11e9-9d51-bf2d952c2a9c.png)

This seems very inefficient for three reasons:

1. It show some games twice
2. It hides a lot of games behind the fold to the right
3. It wastes a lot of empty space below these two rows

I think a full game-grid with sort/filter options with clear installed/not-installed states would be better. A quick mockup:

![image](https://user-images.githubusercontent.com/7945562/53159724-b3ed9780-35c6-11e9-9b4f-2773bad27565.png)

Also there could be more filter or sorting options, maybe even a search bar for the title.

- Sort by last played (with a time last played text in the info line, probably the default)
- Sort by last installed (with a time installed text in the info line)
- Sort by last updated (with a time updated text in the info line)
- Sort by last purchased (with a time purchased text in the info line)
- Sort by genre (with in-between headers for each genre or the genre in the info line)

A quick sketch for the info-line:

![image](https://user-images.githubusercontent.com/7945562/53159229-bc919e00-35c5-11e9-8cad-762667e3efa8.png)

**Sidenote**

I also think the individual game-pictures could be bigger and feel a bit more like "game-boxes". This would also make it possible to display some more infos like name and description, play time within the box (instead on hover but that's a question of taste).

To be honest, I've looked at Steam, Origin and the Epic Games Launcher and I don't think any of them really do it well. I don't think it's an easy problem to solve either. Just sort of started talking about it while I was at it.

**Comment by Gaboris** [comment]
I support this idea wholeheartedly!

Let us see all our games in a vertically scrolling grid and use "tags" to show states and stats. :)

**Comment by fasterthanlime** [comment]
@Democide have you ever clicked on "Installed", "Owned", or either of the "View all" links?

(I ask because it doesn't come up at all in your suggestion)

**Comment by Gaboris** [comment]
Yeah that is fine as an "expanded/list view" besides this grid idea that could be toggled like in Steam, but with better info displays, but I guess they meant it as the default view since this "owned and installed" lists page kinda feels pointless to click twice for my games especially when I don't have any games installed or even just a few.

Heck, just look at GoG's library. It's still not perfect, but a good enough base.

**Comment by Democide** [comment]
@fasterthanlime: I have seen it, but forgot about it when writing the issue. Good point bringing it up.

I think that list is perfectly fine but it feels more like the itch dashboard than a games list since each entry takes up so much space (with a lot of of empty space for each entry.) If the purpose is to provide a quick overview with details, then it fails because it doesn't provide a lot of entries at one glance. Compared to the grid view it doesn't really provide much in the way of advantages. The hover-text is clearly visible, and if I look closely (in the owned version) I can discern between "launch" and install. 

But you're right - I'm pretty much advocating for that to be the default view but with boxes instead of a list. Or maybe with display options, in a sense.

I basically just want to click on Library and have a useful view where I can quickly see what I own, and what I can install or play right now. Maybe with highlights for "recently purchased" so I can install things. And/or recently installed/played so I can launch things.

Quick mockup for list-view with grid option:

![image](https://user-images.githubusercontent.com/7945562/53166471-a12e8f00-35d5-11e9-9d09-ba901d256dc5.png)

![image](https://user-images.githubusercontent.com/7945562/53166477-a390e900-35d5-11e9-8931-c7e7f647f60a.png)

(Obviously the installed/non-installed state should be visible in the same way in the list...)

Tho I have to agree, the massive filter bar up there is a bit unseemly.

AAANYWAY: I feel like taking a page out of folder views would be useful here: a list with small text and a lot of things at one glance is useful. As is a (small / big) thumbnaily-view. Anything in between isn't really seeing much use. At least not on my end. To wit:

![image](https://user-images.githubusercontent.com/7945562/53167011-08007800-35d7-11e9-858b-8e1b4e69cb10.png)

Which makes me think of the clever scale slider in the unity project view. If it's set to minimal, you get a list, if you move it toward the maximum you get a g rid of ever growing thumbnails.

**Comment by Democide** [comment]
Also, a super-tiny nitpick: on the explore tab, the left hand sidebar and the right hand content have different backgrounds (Content is a TAD darker), on the library it's the same color.

**Comment by fasterthanlime** [comment]
> Also, a super-tiny nitpick: on the explore tab, the left hand sidebar and the right hand content have different backgrounds (Content is a TAD darker), on the library it's the same color.

Please open a separate issue for that one - Explore's background is the website's background, which I guess we could unify one way or the other.

---

I agree that the Library tab as it is currently sorta sucks. I do want to move back to something more like v23's style:

![](https://cloud.githubusercontent.com/assets/7998310/16583085/7702c448-42b3-11e6-949a-c5b45e906807.png)

A lot of people have also asked for a list style - v23 had a list, sort of, although the lines where more than 1 line of text high, because it still container a cover and some additional info.

Re the current "list view", ie. "Owned games" and "Installed games", it big and wasteful because I was hoping to eventually grab more information from games and show it there: stuff like screenshots (maybe your screenshots), reviews, tags, etc. I still think a view like that has a place, but maybe it's not a good default (it might be a good default for collections?)

Re having different choices like: details, grid, list, etc. - one problem I keep having is I don't know where to store preferences. When you switch to details, do you only switch for that particular session/tab (you can still enable tabs in v25 right now), for that particular view ("owned games" or "installed games") or is it for all views ever?

**Comment by Democide** [comment]
Yeah, the v23 look is nice. I'd prefer that, clearly. I also really like the install/launch button. However I think you don't need to repeat the title, since it's in most of the covers.

And for choices: I think have it behave how you'd expect a browser to behave. So if you change the display type, it changes the current tab for that "menu-option" (library, collections...) and all new tabs (of that option) from that point on. Existing tabs are unchanged. (see mute button on YouTube for example). 

And the detailed list sounds interesting for collections. I could see that.

**Comment by aaronfranke** [comment]
I agree that this needs to be changed and that v23's style was better, but it isn't great either. 

My suggestions:

* Instead of "Owned" and "Installed", change to "Installed" and "Not Installed" or similar. Put the "Installed" category on top. I do like the idea of separating installed and uninstalled items, unlike Steam's grid view, which puts them all together.

* As @Democide suggested, just make the grid continue downwards. Remove the "View all" button entirely and show all games by default.

* Currently, when left clicking on a game, it opens the game page. Instead of this, I propose that when you hover over a game, it should show "Install" and "Details" for uninstalled games, and "Play" and "Details" for installed games. Clicking "Details" would open the game's Itch page. This is basically what Steam does, it looks like this:

![1](https://user-images.githubusercontent.com/1646875/56466091-beb38580-63d9-11e9-8d0c-b6b377503cf7.png)

But this leaves most of the image useless, since only the blue area is clickable. Instead, I propose that Itch's buttons look more like this (obviously not red of course):

![s](https://user-images.githubusercontent.com/1646875/56466099-d723a000-63d9-11e9-9ac9-dbc9bedb9873.png)

The game could be uninstalled through the page accessed via "Details", or the right-click menu.

**Comment by RavensKrag** [comment]
Are there still plans to implement a grid view like in v23?

**Comment by VRArt1** [comment]
I'd also prefer a grid view.
